operations/http.go

```diff
@@ -605,6 +606,31 @@ func (ac *legacyClient) PutPatch(incomingPatch patchSubmission) (*patch.Patch, e
 	return reply.Patch, nil
 }
 
+func (ac *legacyClient) GetRawPatchWithModules(patchId string) (data.RawPatch, error) {
+
+	rp := data.RawPatch{}
+	resp, err := ac.get2(fmt.Sprintf("patches/%s/raw_modules", patchId), nil)
+
+	if err != nil {
+		return rp, errors.Wrap(err, "making request to get raw patch")
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusUnauthorized {
+		return rp, NewAuthError(resp)
+	}
+	if resp.StatusCode != http.StatusOK {
+		return rp, NewAPIError(resp)
+	}
+
+	err = utility.ReadJSON(resp.Body, &rp)
```

Comment:
nit: you can combine L627-628 into one `if err :=`.

Reply:
done